### DISCUSSION TITLE
Add host-id to IMPLANT-METADATA

### DISCUSSION THREAD

**Discussion by 0x4156-AV** [rasta-mouse/OST-C2-Spec#5] [Ideas]
The IMPLANT-METADATA message currently includes various host identifiers and system information, but lacks a unique, persistent identifier for the host machine.

## Proposed Change:
Add a new field `host-id` to the IMPLANT-METADATA message:

```
IMPLANT-META {
  ...
  host-id        [14]  String                   OPTIONAL
  ...
}
```

## Rationale:
1. **Improved Persistence**: The host GUID or machine-id provides a persistent* identifier that remains constant across reboots and network changes, unlike hostname or IP addresses
2. **Compatibility**: Most OSs provide methods to retrieve this unique machine identifier with no privleges required
3. **Optional Field**: Making this field optional accounts for situations where the machine-id may not be retrievable (Docker containers, for example), or scenarios where transmitting this identifier might not be desirable

**Comment by rasta-mouse** [comment]
Implemented in 722f2f4
